virt-spawn

```diff
@@ -247,9 +247,11 @@ $SUDO virt-builder $DISTRO \
   $BUILD_OPTS \
   || exit 2
 
+CORES_TOTAL=$(grep -c ^processor /proc/cpuinfo)
```

Comment:
There's also /usr/bin/nproc (from coreutils) which does exactly what you want. It's available on EL6.